src/libYARP_os/src/yarp/os/LogStream.h

```diff
@@ -247,26 +249,80 @@ class YARP_os_API LogStream
         return *this;
     }
 
-    template <typename T>
-    inline LogStream& operator<<(const std::vector<T>& t)
+#ifndef SWIG_PREPROCESSOR_SHOULD_SKIP_THIS
+    template <typename T1, typename T2>
+    inline LogStream& operator<<(const std::pair<T1, T2>& t)
+    {
+        stream->oss << '{' << t.first << ", " << t.second << '}';
+        if (!stream->nospace) {
+            stream->oss << ' ';
+        }
+        return *this;
+    }
+
+    template <typename Container, typename = typename Container::value_type>
+    inline LogStream& operator<<(const Container& cont)
```

Comment:
Perhaps this should be a C++17 template template parameter? i.e. something like;
```c++
template<typename T, typename Alloc = std::allocator<T>, template <typename, typename> class Container>
inline LogStream& operator<<(const Container<T, Alloc>& t)
```

Reply:
I'm not sure how this is related to C++17 specifically (perhaps type deduction?), I lack a proper knowledge of this standard yet. It believe it was already achievable with older standards per [this SO answer](https://stackoverflow.com/a/213811/10404307). It seems to me the less convolution, the better (see UPDATE section). In addition, [`std::set<Key, Compare, Allocator>`](https://en.cppreference.com/w/cpp/container/set), among others, might not work.